index.js

```diff
@@ -26,9 +28,10 @@ var GROUP = new RegExp([
   '<([^>]*)>', // We capture the actual range expression
   ')?',
   // The trailing part is optional
-  '([^,;\\s]+)?',
+  '([^,;\\s]*)',
   ')', // end of actual group
 ].join(''), 'g');
+var HOST = /^([^\d,;\s<>]+)([\d]*)([^,;\s<>]*)$/;
```

Comment:
Superfluous `[]` around the `\d` in the middle?  Did you want to explicitly fail on double-numbered hosts? (`foo14sub3.tld`)